repos/system_upgrade/el8toel9/actors/nisscanner/libraries/nisscan.py

```diff
@@ -0,0 +1,60 @@
+import os
+
+from leapp.libraries.stdlib import api
+from leapp.models import NISConfig
+
+PACKAGES_NAMES = ('ypserv', 'ypbind')
+YPBIND_CONF_FILE = '/etc/yp.conf'
+YPSERV_DIR_PATH = '/var/yp'
+YPSERV_DEFAULT_FILES = ('binding', 'Makefile', 'nicknames')
+
+
+class NISScanLibrary(object):
+    """
+    Helper library for NISScan actor.
+    """
+
+    def check_client_configuration(self):
+        """
+        Check for any significant ypbind configuration lines in .conf file.
+        """
+        if not os.path.isfile(YPBIND_CONF_FILE):
+            return False
+
+        # Filter whitespaces and empty lines
+        lines = [line.strip() for line in open(YPBIND_CONF_FILE).readlines() if line.strip()]
+
+        for line in lines:
+            # Cheks for any valid configuration entry
+            if not line.startswith('#'):
+                return True
+        return False
+
+    def check_server_configuration(self):
```

Comment:
check_server_configuration name implies some validation, but iiuc the idea is to check if any additional config files are present. Maybe consider renaming it to something more self-speaking, like `has_non_default_configuration` or `check_non_default_configuration`?